mne/time_frequency/tfr.py

```diff
@@ -878,15 +878,34 @@ def crop(self, tmin=None, tmax=None):
             Start time of selection in seconds.
         tmax : float | None
             End time of selection in seconds.
+        fmin : float | None
+            Lowest frequency of selection in Hz.
+
+            .. versionadded:: 0.18.0
+        fmax : float | None
+            Highest frequency of selection in Hz.
+
+            .. versionadded:: 0.18.0
 
         Returns
         -------
         inst : instance of AverageTFR
             The modified instance.
         """
-        mask = _time_mask(self.times, tmin, tmax, sfreq=self.info['sfreq'])
-        self.times = self.times[mask]
-        self.data = self.data[..., mask]
+        if any((tmin, tmax)):
+            time_mask = _time_mask(self.times, tmin, tmax,
+                                   sfreq=self.info['sfreq'])
+
+            self.times = self.times[time_mask]
+            self.data = self.data[..., time_mask]
+
+        if any((fmin, fmax)):
+            freq_mask = _freq_mask(self.freqs, sfreq=self.info['sfreq'],
+                                   fmin=fmin, fmax=fmax)
+
+            self.freqs = self.freqs[freq_mask]
+            self.data = self.data[..., freq_mask, :]
```

Comment:
I think we can do even more efficient

```
        if any((tmin, tmax)):
            time_mask = _time_mask(self.times, tmin, tmax,
                                   sfreq=self.info['sfreq'])
       else:
            time_mask = slice()

        if any((fmin, fmax)):
            freq_mask = _freq_mask(self.freqs, sfreq=self.info['sfreq'],
                                   fmin=fmin, fmax=fmax)
        else:
            freq_mask = slice()

       self.times = self.times[time_mask]
       self.freqs = self.freqs[freq_mask]
       self.data = self.data[..., freq_mask, time_mask]
```

Reply:
with this you do in any case only one copy

Reply:
👍

Reply:
This code will fail for (at least) `tmin=0, tmax=0`. Yes the fix is easy, but my point is if we want to optimize this, let's do it separately and make sure we don't break anything in new ways. The current PR is based on code that has been tweaked a lot over the years to handle real user edge cases. Let's get this PR in and optimize later.

Reply:
why do you say it will fail?

Reply:
any((0, 0)) will be False so it won't actually crop to (0, 0), which is what `tmin=0, tmax=0` should do. Your check should be `if tmin is not None or tmax is not None`.

But you see my point about this? I really think this discussion should be in a subsequent PR, not here.

Reply:
(Yes the current code is inefficient, but so is `_time_mask` so we should optimize both later in a PR where we can focus on this properly.)

Reply:
I'd do it like this now:
```python
        if tmin is not None or tmax is not None:
            time_mask = _time_mask(self.times, tmin, tmax,
                                   sfreq=self.info['sfreq'])

        else:
            time_mask = slice(None)

        if fmin is not None or fmax is not None:
            freq_mask = _freq_mask(self.freqs, sfreq=self.info['sfreq'],
                                   fmin=fmin, fmax=fmax)
        else:
            freq_mask = slice(None)

        self.times = self.times[time_mask]
        self.freqs = self.freqs[freq_mask]
        self.data = self.data[..., freq_mask,:][...,time_mask]
```
Since ```self.data = self.data[..., freq_mask, time_mask]``` leads to wrong indexing for e.g. ```tfr.crop(tmin=0.3, tmax=0.4, fmin=0.1, fmax=0.2)``` and using ```np.ix_``` i can't find another way of satisfying all possible combinations of slices/indexes.

Would it be okay like this or can anyone perhaps help me out here?